pkg/apiserver/profiling/model.go

```diff
@@ -49,7 +65,8 @@ func (TaskModel) TableName() string {
 
 type TaskGroupModel struct {
 	ID                  uint                          `json:"id" gorm:"primary_key"`
-	State               TaskState                     `json:"state" gorm:"index"`
+	State               GroupTaskState                `json:"state" gorm:"index"` // legacy state, works before version v2021.11.01.1
+	NewState            GroupTaskState                `json:"new_state" gorm:"index"`
```

Comment:
Do we really need `NewState`? As [design docs comments](https://docs.google.com/document/d/1O4j8H04GJ8Iyg3jVPjQliRZhUfQ4CWHJHvrJhdFKh6Y/edit?disco=AAAARp92vmQ) said, I'm worried that having two such similar fields, `State` and `NewState`, will confuse developers.

Reply:
Yep, I agree with you, but it seems only this way can handle the situation when down-grading.